common/src/main/java/com/zegelin/cassandra/exporter/FactoriesSupplier.java

```diff
@@ -179,6 +181,27 @@ private Factory clientMetricFactory(final FactoryBuilder.CollectorConstructor co
                 .build();
     }
 
+
+    private Factory clientStreamingMetricFactory(final FactoryBuilder.CollectorConstructor collectorConstructor, final String jmxName, final String familyNameSuffix, final String help) {
+        final ObjectName objectNamePattern = format("org.apache.cassandra.metrics:type=Streaming,name=%s,scope=*", jmxName);
+        final String metricFamilyName = String.format("stream_%s", familyNameSuffix);
+
+        return new FactoryBuilder(collectorConstructor, objectNamePattern, metricFamilyName)
+                .withHelp(help)
+                .withModifier((keyPropertyList, labels) -> {
+                    final String scope = keyPropertyList.get("scope");
+                    labels.put("partner_endpoint", scope);
+                    try {
```

Comment:
To simplify a bit this method, we could extract this into a method:

```java
private Optional<String> getHostname() {
    try {
        return Optional.of(InetAddress.getByName(scope).getHostName()));
    } catch (UnknownHostException e) {
        return Optional.empty();
    }
}
```

Then here we could put something like: 
```java
getHostname().ifPresent(hostname -> labels.put("partner_hostname", hostname));
```

Reply:
Done